src/test/java/rx/ObservableTests.java

```diff
@@ -1101,19 +1105,45 @@ public String call(Integer t1) {
     }
     
     @Test
-    public void testErrorThrownIssue1685() {
+    public void testErrorThrownIssue1685() throws ExecutionException, InterruptedException {
         Subject<Object, Object> subject = ReplaySubject.create();
 
-        Observable.error(new RuntimeException("oops"))
-            .materialize()
-            .delay(1, TimeUnit.SECONDS)
-            .dematerialize()
-            .subscribe(subject);
+        ExecutorService exec = Executors.newSingleThreadExecutor();
 
-        subject.subscribe();
-        subject.materialize().toBlocking().first();
+        try {
+
+            final AtomicReference<Throwable> err = new AtomicReference<Throwable>();
+
+            Scheduler s = Schedulers.from(exec);
+            exec.submit(new Runnable() {
+                @Override
+                public void run() {
+                    Thread.currentThread().setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+                        @Override
+                        public void uncaughtException(Thread t, Throwable e) {
+                            err.set(e);
+                        }
+                    });
+                }
+            }).get();
 
-        System.out.println("Done");
+            Observable.error(new RuntimeException("oops"))
+                    .materialize()
+                    .delay(1, TimeUnit.SECONDS, s)
+                    .dematerialize()
+                    .subscribe(subject);
+
+            subject.subscribe();
+            subject.materialize().toBlocking().first();
+
+            Thread.sleep(1000); // the uncaught exception comes after the terminal event reaches toBlocking
```

Comment:
This fails for some reason. Maybe the underlying thread dies off?
Could you change the code above to use a `ThreadFactory` with `newSingleThreadExecutor` and set the uncaught exception handler within the factory?